Automattic/wp-calypso
# Enable eslint rule `import/no-extraneous-dependencies` for all packages

### Background

The rule `import/no-extraneous-dependencies` will allow us to have a better defined dependency tree, where each package explicitly define its dependencies and don't rely on the dependencies of the root package. However, we are not enforcing the eslint rule `import/no-extraneous-dependencies` in our repo. 

Some packages (mostly those inside `./packages`) do enable it, but not all. Furthermore, some files have declared exceptions to the rule for specific imports. This creates some inconsistency on how we deal with dependencies.

### Changes

This PR: 

* Enables the rule `import/no-extraneous-dependencies` for all `./packages`
* Removes exceptions and customizations related to this rule
* Add missing dependencies where necessary to prevent violations of this rule.

### Testing instructions

* Run `./node_modules/.bin/eslint-nibble --ext .js,.jsx,.ts,.tsx,.md.js,.md.javascript,.md.jsx packages`, all errors related to `import/no-extraneous-dependencies` should be gone.
* Verify unit tests still pass.
